examples/threading/threading.c

```diff
@@ -0,0 +1,128 @@
+#include "threading.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <errno.h>
+
+#define MILLI_TO_MICRO_SCALE (1000)
+
+// Optional: use these functions to add debug or error prints to your application
+#define DEBUG_LOG(msg,...)
+//#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+#define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
+
+void* threadfunc(void* thread_param)
+{
+
+    // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
+    // hint: use a cast like the one below to obtain thread arguments from your parameter
+    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    int err_status = 0;
+    struct thread_data *thread_func_args = (struct thread_data *) thread_param;
+
+    // wait before obtaining mutex
+    int ret = usleep(thread_func_args->wait_to_obtain_ms * MILLI_TO_MICRO_SCALE);
+    if( ret == -1 )
+    {
+        ERROR_LOG("Error usleep at obtain failed with error number %d", errno);
+        err_status = 1;
+    }
+    else
+    {
+        DEBUG_LOG("Obtain mutex sleep success!!");
+    }
+
+
+    // acquire mutex
+    ret = pthread_mutex_lock(thread_func_args->mtx);
+    if( ret == -1 )
+    {
+        ERROR_LOG("Error mutex lock failed with error number %d", errno);
+        err_status = 1;
+    }
+    else
+    {
+        DEBUG_LOG("Obtain mutex success!!");
+    }
+
+
+    // wait before release of mutex
+    ret = usleep(thread_func_args->wait_to_release_ms * MILLI_TO_MICRO_SCALE);
+    if( ret == -1 )
+    {
+        ERROR_LOG("Error usleep at release failed with error number %d", errno);
+        err_status = 1;
+    }
+    else
+    {
+        DEBUG_LOG("Release mutex sleep success!!");
+    }
```

Comment:
Good!  I think the debug print is wrong here though.

Reply:
I intended it to mean the wait before release of mutex.